beamer/tests/agent/multiple_processors/test_multiple_event_processors.py

```diff
@@ -0,0 +1,224 @@
+    return psutil.Popen(
+        ["ape", "test", str(SLAVE_TEST_PATH), "-s"],
+        bufsize=0,
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        env=dict(os.environ, PORT=str(port), REQUEST_COUNT=str(request_count)),
+    )
+
+
+@contextlib.contextmanager
+def _new_networks(chain_port_map: dict[ChainId, int]):
+    processes = {}
+    try:
+        for chain_id, port in chain_port_map.items():
+            processes[chain_id] = _start_ganache(chain_id, port)
+        yield
+    finally:
+        for process in processes.values():
+            os.kill(process.pid, signal.SIGKILL)
+            process.wait()
+
+
+def _get_chain_map() -> dict[ChainId, int]:
+    number_of_chains = 3
+    return {ChainId(i): i for i in range(8546, 8546 + number_of_chains)}
+
+
+def _get_config(
+    chain_map: dict[ChainId, int],
+    contracts: Contracts,
+    slave_contract_addresses: dict[ChainId, dict[str, str]],
+) -> Config:
+    account = eth_account.Account.from_key(_LOCAL_ACCOUNT.private_key)
+    url = ape.config.provider.uri
+    deployment_info = {}
+
+    for chain_id in chain_map:
+        contracts_info = dict(
+            RequestManager=ContractInfo(
+                deployment_block=BlockNumber(1),
+                address=to_canonical_address(
+                    slave_contract_addresses[chain_id]["request_manager"]
+                ),
+                abi=[abi.dict() for abi in contracts.request_manager.contract_type.abi],
+            ),
+            FillManager=ContractInfo(
+                deployment_block=BlockNumber(1),
+                address=to_canonical_address(slave_contract_addresses[chain_id]["fill_manager"]),
+                abi=[abi.dict() for abi in contracts.fill_manager.contract_type.abi],
+            ),
+        )
+        deployment_info[chain_id] = contracts_info
+    rpc_urls = {"l1": url}
+    for chain_id, port in chain_map.items():
+        rpc_url = f"http://127.0.0.1:{port}"
+        rpc_urls[str(chain_id)] = rpc_url
+
+    token_list = _get_token_list(chain_map, slave_contract_addresses)
+    config = Config(
+        rpc_urls=rpc_urls,
+        deployment_info=deployment_info,
+        token_checker=TokenChecker(token_list),
+        account=account,
+        fill_wait_time=0,
+        unsafe_fill_time=600,
+        prometheus_metrics_port=None,
+        log_level="debug",
+    )
+    return config
+
+
+def _start_agent_test(config: Config):
+    agent = Agent(config)
+    agent.start()
+
+    directions = agent.get_directions()
+    try:
+        for direction in directions:
+            with Sleeper(20) as sleeper:
+                while len(agent.get_context(direction).requests) != 1:
+                    sleeper.sleep(0.1)
+
+            request = tuple(agent.get_context(direction).requests)[0]
+
+            with Sleeper(20) as sleeper:
+                while not request.is_claimed:
+                    sleeper.sleep(0.1)
+    finally:
+        agent.stop()
```

Comment:
`agent.wait` is not necessary here because `agent.stop` will anyway block due to `event_processor.stop` and `event_monitor.stop` both being blocking, right?

Reply:
yes.